pallets/cash/src/internal/set_yield_next.rs

```diff
@@ -15,48 +15,38 @@ use types_derive::Types;
 
 #[derive(Copy, Clone, Eq, PartialEq, Encode, Decode, Debuggable, Types)]
 pub enum SetYieldNextError {
-    TimestampTooSoonToNow,
-    TimestampTooSoonToNext,
+    NotEnoughTimeToSyncBeforeNext,
+    NotEnoughTimeToSyncBeforeCancel,
 }
 
-fn get_cash_yield_index_after<T: Config>(change_in_time: Timestamp) -> Result<CashIndex, Reason> {
+fn get_cash_yield_index_after<T: Config>(dt: Timestamp) -> Result<CashIndex, Reason> {
     let cash_yield = CashYield::get();
     let cash_index_old = GlobalCashIndex::get();
-    let increment = cash_yield.compound(change_in_time)?;
-    cash_index_old
-        .increment(increment.into())
-        .map_err(Reason::MathError)
+    let increment = cash_yield.compound(dt)?;
+    Ok(cash_index_old.increment(increment.into())?)
 }
 
 pub fn set_yield_next<T: Config>(
     next_yield: APR,
     next_yield_start: Timestamp,
 ) -> Result<(), Reason> {
-    let now = get_now::<T>();
-    let change_in_time = next_yield_start
-        .checked_sub(now)
-        .ok_or(Reason::TimeTravelNotAllowed)?;
+    let now = get_recent_timestamp::<T>()?;
+    let min_t = now + MIN_NEXT_SYNC_TIME;
```

Comment:
We're okay with the unchecked addition since it's a) just a check, and b) the paramis a const?

Reply:
Yeah I was thinking about that too, I didn't change it though. I guess the expectation is that current timestamp + MIN_NEXT_SYNC_TIME must fit into a u64?